services/datamanager/builtin/sync/upload_data_capture_file.go

```diff
@@ -0,0 +1,226 @@
+package sync
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/docker/go-units"
+	"github.com/go-viper/mapstructure/v2"
+	"github.com/pkg/errors"
+	v1 "go.viam.com/api/app/datasync/v1"
+	pb "go.viam.com/api/component/camera/v1"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"go.viam.com/rdk/data"
+	"go.viam.com/rdk/logging"
+)
+
+// MaxUnaryFileSize is the max number of bytes to send using the unary DataCaptureUpload, as opposed to the
+// StreamingDataCaptureUpload.
+var MaxUnaryFileSize = int64(units.MB)
+
+// uploadDataCaptureFile uploads the *data.CaptureFile to the cloud using the cloud connection
+// returns context.Cancelled if ctx is cancelled before upload completes.
+// If f is of type BINARY_SENSOR and it's size is over MaxUnaryFileSize,
+// uses StreamingDataCaptureUpload API so as to not exceed the unary response size.
+// Otherwise uploads data over DataCaptureUpload API.
```

Comment:
```suggestion
// uploadDataCaptureFile uploads the *data.CaptureFile to the cloud using the cloud connection.
// returns context.Cancelled if ctx is cancelled before upload completes.
// If f is of type BINARY_SENSOR and its size is over MaxUnaryFileSize,
// uses StreamingDataCaptureUpload API so as to not exceed the unary response size.
// Otherwise, uploads data over DataCaptureUpload API.
```